infra/src/main/kotlin/fi/fta/geoviite/infra/publication/PublicationValidation.kt

```diff
@@ -40,36 +40,43 @@ fun validateTrackNumberReferences(
     referenceLine: ReferenceLine?,
     kmPosts: List<TrackLayoutKmPost>,
     locationTracks: List<LocationTrack>,
```

Comment:
Täällä on joitain loogisia muutoksia:
- Poistettu tarpeettomat yritykset validoida että joku muu joka viittaa validoitavaan olioon ei ole draft-only. Tuolla ei pitäisi olla koskaan väliä.
- Poistettu null vs not-published eroja: uuden kontekstin myötä not-published näkyy siten että viitattu olio on null
- Siirretty joitain publicationservicen puolella olleita validointeja tänne koska ne oli nyt helpompaa tehdä puhtaina funkkareina